### PR TITLE
Do not remove azure additional info when removing instance

### PR DESCRIPTION
**Pull request by mbenguig**

### CODE REVIEW

File: infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java [hunk 1]
```diff
@@ -436,42 +436,6 @@ private void initBilling() {
...
-                        connectorIaasController.terminateInstance(infrastructureId, instanceId);
-                        this.nodeSource.removeAndPersistAdditionalInformation(CLOUD_COST_RESOURCE_USAGE_REPORTED_AT_KEY,
-                                                                              CLOUD_COST_RESOURCE_USAGE_REPORTED_UNTIL_KEY,
-                                                                              CLOUD_COST_GLOBAL_COST_KEY,
-                                                                              CLOUD_COST_CURRENCY_KEY,
-                                                                              CLOUD_COST_MAX_BUDGET_KEY,
-                                                                              CLOUD_COST_GLOBAL_COST_IN_MAX_BUDGET_PERCENTAGE_KEY);
-                        LOGGER.info("Instance terminated: " + instanceId);
-                    }
-                    nodesPerInstance.remove(instanceId);
```
Comment (mcompastie) [inline infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java]: Shouldn't the nodePerInstance data structure removed when the instance is removed ?

File: infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java [hunk 2]
```diff
@@ -436,42 +436,6 @@ private void initBilling() {
...
-        setPersistedInfraVariable(() -> {
-            // first read from the runtime variables map
-            nodesPerInstance = (Map<String, Set<String>>) persistedInfraVariables.get(NODES_PER_INSTANCES_KEY);
-            // make modifications to the nodesPerInstance map
-            if (nodesPerInstance.get(instanceId) != null) {
-                nodesPerInstance.get(instanceId).remove(nodeName);
-                LOGGER.info("Removed node: " + nodeName);
-                if (nodesPerInstance.get(instanceId).isEmpty()) {
-                    if (terminateInstanceIfEmpty) {
-                        connectorIaasController.terminateInstance(infrastructureId, instanceId);
```
Comment (mcompastie) [inline infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java]: Shouldn't the instance be terminated ?

Comment (mbenguig) [reply infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java]: This method already exists in the mother class.

File: infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java (hunk 1 above)
Comment (mbenguig) [reply infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java]: This method already exists in the mother class.

**Pull request merged**